packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -288,8 +296,22 @@ export default function CollabSidebar() {
 			}
 		} );
 
-		return result;
-	}, [ threads ] );
+		if ( 0 === result?.length ) {
+			return { resultComments: [], sortedThreads: [] };
+		}
+
+		const blockCommentIds = getCommentIdsFromBlocks( blocks );
+
+		const threadIdMap = new Map(
+			result?.map( ( thread ) => [ thread.id, thread ] )
+		);
+
+		const sortedComments = blockCommentIds
+			.map( ( id ) => threadIdMap.get( id ) )
```

Comment:
Btw, I see above that a thread can be `{ ...item, reply: [] }`, which will cause this component to constantly re-render.

Reply:
I don't follow, we are re-arranging comments here so reply comments will go under the parent comment.

Reply:
Sorry, I missed that this is within `useMemo` instead of `useSelect`